lib/rules/prefer-const.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @fileoverview A rule to suggest using of const declaration for never changed variables.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    /**
+     * Checks whether a reference is the initializer.
+     * @param {Reference} reference - A reference to check.
+     * @returns {boolean} Whether or not the reference is the initializer.
+     */
+    function isInitializer(reference) {
+        return reference.init === true;
+    }
+
+    /**
+     * Checks whether a reference is read-only or the initializer.
+     * @param {Reference} reference - A reference to check.
+     * @returns {boolean} Whether or not the reference is read-only or the initializer.
+     */
+    function isReadOnlyOrInitializer(reference) {
+        return reference.isReadOnly() || reference.init === true;
+    }
+
+    /**
+     * Search and report never changed variables.
+     * @param {Scope} scope - A scope of the search domain.
+     * @returns {void}
+     */
+    function checkForNeverChangedVariables(scope) {
+        // Skip the TDZ type.
+        if (scope.type === "TDZ") {
+            return;
+        }
+
+        var variables = scope.variables;
+        for (var i = 0, end = variables.length; i < end; ++i) {
+            var variable = variables[i];
+            var def = variable.defs[0];
+            var declaration = def && def.parent;
+            var statement = declaration && declaration.parent;
+            var references = variable.references;
+            var identifier = variable.identifiers[0];
+
+            if (statement != null &&
+                identifier != null &&
+                declaration.type === "VariableDeclaration" &&
+                declaration.kind === "let" &&
+                (statement.type !== "ForStatement" || statement.init !== declaration) &&
+                (statement.type !== "ForInStatement" || statement.left !== declaration) &&
+                references.some(isInitializer) &&
+                references.every(isReadOnlyOrInitializer)
+            ) {
+                context.report(
+                    identifier,
+                    "`{{name}}` is never changed, use `const` instead.",
+                    {name: identifier.name});
+            }
+        }
+    }
+
+    var pushAll = Function.apply.bind(Array.prototype.push);
```

Comment:
Because I don't like a notation such as `stack.push.apply(stack, scope.childScopes)`, I wanted to define a function:

```js
function pushAll(stack, values) {
    stack.push.apply(stack, values);
}
```

`Function.apply.bind(Array.prototype.push)` is same as that function -- `pushAll`.